### DISCUSSION TITLE
Supabase secret not working

### DISCUSSION THREAD

**Discussion by Kgitman** [supabase/supabase#28535] [Questions]
HEY, @GaryAustin1 and Supabase team!

Thanks for seeing this!

I deployed my edge function with Supabase CLI and I used that URL to set the cron job 
with this sql code : 

```
select
  cron.schedule(
    'invoke-function-every-minute',
    '* * * * *', -- every minute
    $$
    select
      net.http_post(
        url := 'https://jphyclgetgrgcslebwqa.supabase.co/functions/v1/my_function',
        headers := jsonb_build_object(
          'Content-Type', 'application/json',
              'Authorization', 'Bearer ' || current_setting('SUPABASE_ANON_KEY')
        ),
        body := concat('{"time": "', now(), '"}')::jsonb
      ) as request_id;
    $$
  );
```
  
  BUT my cron job failing some time with ERROR : 
  
  **ERROR** :  unrecognized configuration parameter "SUPABASE_ANON_KEY"

  or some time 401 response
  
  I checked the secret and is all okay.
  
  **MY edge function is something like this** : 
  
  IN `index.ts` I imported my supabase credentials which was stored in `supabase_client.ts` file.
  
  `import { getSupabaseClient } from './supabase_client.ts';`
  
  and my `supabase_client.ts` file is like this : 
  
```
  import { createClient } from "https://esm.sh/@supabase/supabase-js@2.0.0";
import "jsr:@std/dotenv/load";

// Supabase credentials
const SUPABASE_URL = Deno.env.get("SUPABASE_URL") as string;
const SUPABASE_ANON_KEY = Deno.env.get("SUPABASE_ANON_KEY") as string;

// Create a Supabase client
const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);

export function getSupabaseClient() {
  return supabase;
}
```
  
  I tried many things but the cron job is not working!
  
  
  when my Supabase credentials are in main file like `index.ts` at that time the cron job work perfectly but when I import the credentials from the .env files at that time it doesn't work!
  
  I used the secret but it's not work!
  
  It's great if you got me out of this!
  
  
  **If you want any information than I will ready to give you that!**

**Comment by GaryAustin1** [comment]
To my knowledge the keys are not available in the database.  You will need to hardcode it, or better yet use Vault to store it and then select it from Vault in your cron code.

**Reply by Kgitman** [reply]
well, I will try!
